dlt/helpers/airflow_helper.py

```diff
@@ -308,15 +316,53 @@ def log_after_attempt(retry_state: RetryCallState) -> None:
                 if pipeline.full_refresh:
                     raise ValueError("Cannot decompose pipelines with full_refresh set")
 
-                # parallel tasks
                 tasks = []
                 sources = data.decompose("scc")
+                t_name = task_name(pipeline, data)
                 start = make_task(pipeline, sources[0])
 
+                # parallel tasks
                 for source in sources[1:]:
+                    for resource in source.resources.values():
+                        if resource.incremental:
+                            logger.warn(
+                                (
+                                    f"The resource {resource.name} in task {t_name} "
+                                    "is using incremental loading and may modify the "
+                                    "state. Resources that modify the state should not "
+                                    "run in parallel within the single pipeline as the "
+                                    "state will not be correctly merged. Please use "
+                                    "'serialize' or 'parallel-isolated' modes instead."
+                                )
+                            )
+                            break
                     tasks.append(make_task(pipeline, source))
 
-                end = DummyOperator(task_id=f"{task_name(pipeline, data)}_end")
+                end = DummyOperator(task_id=f"{t_name}_end")
+
+                if tasks:
+                    start >> tasks >> end
+                    return [start] + tasks + [end]
+
+                start >> end
+                return [start, end]
+            elif decompose == "parallel-isolated":
+                if not isinstance(data, DltSource):
+                    raise ValueError("Can only decompose dlt sources")
+
+                if pipeline.full_refresh:
+                    raise ValueError("Cannot decompose pipelines with full_refresh set")
+
+                # parallel tasks
+                tasks = []
+                t_name = task_name(pipeline, data)
+
+                start = DummyOperator(task_id=f"{t_name}_start")
+
+                for task_num, source in enumerate(data.decompose("scc"), start=1):
```

Comment:
let's do the same trick and in the mode above. task1 goes first and then all the rest in parallel. first task possibly will create the dataset and initial schema. sorry if that was not clear in the ticket

Reply:
In fact, first I did it, but reverted before committing

Reply:
@IlyaFaer I think your revert is not pushed. otherwise this PR is top